pymc/pytensorf.py

```diff
@@ -111,6 +115,18 @@ def convert_data(data) -> np.ndarray | Variable:
             ret = np.ma.MaskedArray(vals, mask)
         else:
             ret = vals
+    elif hasattr(data, "to_numpy") and hasattr(data, "is_null"):
```

Comment:
```suggestion
    elif hasattr(data, "to_numpy") and hasattr(data, "is_null"):
        # Probably polars object
```

Reply:
Why not a bit more explicit:
```suggestion
    elif pl is not None and isinstance(data, (pl.DataFrame, pl.Series)):
```
The polars namespace is used anyway (in the except clause).